packages/vats/src/vat-transfer.js

```diff
@@ -0,0 +1,67 @@
+// @ts-check
+import { Far } from '@endo/far';
+import { makeDurableZone } from '@agoric/zone/durable.js';
+
+import { prepareBridgeTargetModule } from './bridge-target.js';
+import { prepareTransferTools } from './transfer.js';
+
+export const buildRootObject = (_vatPowers, _args, baggage) => {
+  const zone = makeDurableZone(baggage);
+
+  const { makeBridgeTargetKit } = prepareBridgeTargetModule(
+    zone.subZone('bridge'),
+  );
+
+  const { makeTransferMiddleware } = prepareTransferTools(
+    zone.subZone('transfer'),
+  );
+
+  /**
+   * This 2-level structure is to avoid holding the bridge managers strongly, as
+   * well as mitigate the lack of complex keys.
+   *
+   * @type {WeakMapStore<
+   *   import('./types').ScopedBridgeManager,
+   *   MapStore<string, ReturnType<typeof makeBridgeTargetKit>>
+   * >}
+   */
+  const managerToKits = zone.weakMapStore('managerToHandler');
+  return Far('TransferVat', {
+    /**
+     * @param {import('./types').ScopedBridgeManager} manager
+     * @param {string} [inboundType]
+     */
+    provideBridgeTargetKit(manager, inboundType = 'IBC_EVENT') {
+      /** @type {MapStore<string, ReturnType<typeof makeBridgeTargetKit>>} */
+      let inboundTypeToKit;
+      if (managerToKits.has(manager)) {
+        inboundTypeToKit = managerToKits.get(manager);
+      } else {
+        inboundTypeToKit = zone.detached().mapStore('inboundTypeToKit');
+        managerToKits.init(manager, inboundTypeToKit);
+      }
```

Comment:
More clear with `provideLazy`. Plus it can then be `const` (doesn't ever change) and I think provideLazy infers the type for you
```suggestion
      const inboundTypeToKit = provideLazy(managerToKits, manager, () => zone.detached().mapStore('inboundTypeToKit'));
```